src/stratum_server.cpp

```diff
@@ -1287,59 +1287,60 @@ void StratumServer::api_update_local_stats(uint64_t timestamp)
 	if (!m_apiLastUpdateTime.compare_exchange_strong(t, timestamp)) {
 		return;
 	}
+	
+	m_pool->api()->set(p2pool_api::Category::LOCAL, "stratum",
```

Comment:
You don't need to move `m_pool->api()->set` up, it's better to just add `this` to the lambda capture and only add the loop over all connections. I prefer to keep the git diff minimal for reviews. Also, moving the ReadLock inside can harm performance, because this callback can be called more than one time now.